packages/flutter_tools/lib/src/hot.dart

```diff
@@ -178,26 +178,38 @@ class HotRunner extends ResidentRunner {
       return 2;
     }
 
-    if (observatoryPortCompleter != null && result.hasObservatory)
-      observatoryPortCompleter.complete(result.observatoryPort);
-
     await connectToServiceProtocol(result.observatoryPort);
 
     if (device.needsDevFS) {
+      try {
+        Uri baseUri = await _initDevFS();
+        if (connectionInfoCompleter != null) {
+          connectionInfoCompleter.complete(
+            new DebugConnectionInfo(result.observatoryPort, baseUri: baseUri.toString())
+          );
+        }
+      } catch (error) {
+        printError('Error initializing DevFS: $error');
+        return 3;
+      }
       _loaderShowMessage('Connecting...', progress: 0);
-      bool result = await _updateDevFS(
+      bool devfsResult = await _updateDevFS(
         progressReporter: (int progress, int max) {
-          _loaderShowMessage('Syncing files to device...', progress: progress, max: max);
+          if (progress % 10 == 0)
```

Comment:
This updates the loading bar on the device's screen and we should send each one of these events.